File: piotr/org.uva.sc.piotr.ql/src/ast/model/statements/IfStatement.java
package ast.model.statements;

import ast.model.expressions.Expression;
import ast.visitors.ASTNodeVisitor;

import java.util.ArrayList;
import java.util.List;

public class IfStatement extends Statement {

    private Expression condition;
    private List<Statement> statementList = new ArrayList<>();
    private List<Statement> elseStatementList = new ArrayList<>();

    public IfStatement(Expression condition, List<Statement> statementList, List<Statement> elseStatementList, MetaInformation metaInformation) {
        super(metaInformation);
        this.condition = condition;
        this.statementList = statementList;
        this.elseStatementList = elseStatementList;
    }

    public Expression getCondition() {
        return condition;
    }

    public List<Statement> getElseStatementList() {
        return elseStatementList;
    }

    public List<Statement> getStatementList() {
        return statementList;
    }

    @Override
    public <T> T accept(ASTNodeVisitor<T> visitor) {
        return visitor.visit(this);
    }
}

File: DogeQL/src/main/kotlin/doge/ast/node/IfStatement.kt
package doge.ast.node

import doge.ast.location.SourceLocation
import doge.ast.node.expression.Expression
import doge.visitor.QuestionnaireASTBaseVisitor

class IfStatement(val expression: Expression, val block: Block, location: SourceLocation) : Statement(location) {

    override fun <T> accept(visitor: QuestionnaireASTBaseVisitor<T>): T {
        return visitor.visit(this)
    }

}